File: binary_search.c
#include <stdio.h>

int BinarySearch(int arr[],int l,int r,int x){
    if(r>=l){
        int mid = l+(r-l)/2;
        if (arr[mid]==x)
        {
            return mid;
        }
        else if(arr[mid]>x)
        {
            return BinarySearch(arr,l,mid-1,x);
        }
        else{
            return BinarySearch(arr,mid+1,r,x);   
        }
        
    }
    else{
        return -1;
    }
}

int main(){
    int arr[] = {14, 52, 57, 64, 82, 86, 95};
    int x = 64;
    int n = sizeof(arr)/sizeof(arr[0]);
    int result = BinarySearch(arr,0,n-1,x);
    if(result == -1){
        printf("%d Not Found",x);
    }else{
        printf("%d is at index %d",arr[result],result);
    }
}

File: binary_search.py
def BinarySearch(arr, start, length, find):
    if length >= start:
        
        mid = start+(length-start)//2

        if arr[mid] == find:
            return mid

        elif arr[mid] > find:
            return BinarySearch(arr, start, mid-1, find)

        else:
            return BinarySearch(arr, mid+1, length, find)
    
    else:
        return -1


array = [24,1,5,48,59,65,28,96,94,920]
array.sort()
print(array)
find = 59

#BinarySearch(arr,starting_index,length,finding_elemant)
result = BinarySearch(array,0,len(array)-1,find)

if result == -1:
    print("Not Found")
else:
    print(f"{array[result]} number is at index {result}")